### PR TITLE
Add basic support for custom tags

### PR DESCRIPTION
**Pull request by mauricioabreu**
Main parser function now receives a function that can be
used to parse the line content and save it into the data store object.

### CODE REVIEW

**Comment by mauricioabreu** [comment]
I have another idea here: we can add the "ignored" tags in a new data structure, containing the `line`, `lineno` and any other useful information that can be directly used by `m3u8` users after they parse the content.

```python
...
        elif line.startswith('#'):
            data['ignored_lines'] = {'line': line, 'lineno': lineno}
...

# Usage
m3u8_obj = m3u8.loads('#EXTM3U8 ... etc ... ')
data = post_process_data(m3u8_obj.data)
```

File: tests/test_parser.py [hunk 1]
```diff
@@ -313,3 +313,12 @@ def test_should_parse_start_with_precise():
...
     assert data['discontinuity_sequence'] == 123
+
+def test_simple_playlist_with_custom_tags():
+    def get_movie(line, data, lineno):
+        custom_tag = line.split(':')
+        if len(custom_tag) == 2:
+            data['movie'] = custom_tag[1].strip()
+
+    data = m3u8.parse(playlists.SIMPLE_PLAYLIST_WITH_CUSTOM_TAGS, strict=False, custom_tags_parser=get_movie)
+    assert data['movie'] == 'million dollar baby'
```
Comment (leandromoreira) [inline tests/test_parser.py]: Can you assert also the others fields? (so we can make sure that it won't break the existent parser)

Comment (mauricioabreu) [reply tests/test_parser.py]: Done!

**Comment by leandromoreira** [comment]
LGTM and solves https://github.com/globocom/m3u8/issues/123 and https://github.com/globocom/m3u8/issues/124 does all the tests pass locally @mauricioabreu ? I tried to restart the build manually but the same error happened 

```
 Plugin 'pytest_cov' could not be loaded:
```

**Comment by mauricioabreu** [comment]
@leandromoreira I was reading about this error and it seems to be incompatibility of `python version` vs `pytest_cov` plugin. One way to solve this issue is using `tox`: https://tox.readthedocs.io/en/latest/

**Comment by mauricioabreu** [comment]
Python 3.5 and Python 2.6 passed the test suite: https://travis-ci.org/globocom/m3u8/builds/485545837?utm_source=github_status&utm_medium=notification

**Comment by leandromoreira** [comment]
so I'll merge it and then I'll try to use this tox thing.

**Comment by mauricioabreu** [comment]
@leandromoreira we don't pin versions of any dependence. Recently `pytest_cov` deprecated some python versions as we can see here: https://github.com/pytest-dev/pytest-cov/blob/master/setup.py#L127

```
python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*'
```

Maybe `tox` can help us to define different dependency versions for different python versions.

**Comment by leandromoreira** [comment]
We should do this.

**Comment by leandromoreira** [comment]
good catch

**Pull request merged**